test/integration-tests/ui/PackageDependencyProvider.test.ts

```diff
@@ -13,107 +13,184 @@
-
-        const files = await treeProvider.getChildren(childFolder);
-        const file = files.find(n => n.name === "CAtomic.c");
-        expect(file).to.not.be.undefined;
-
-        expect(file?.path).to.equal(`${path}/Sources/CAtomic/CAtomic.c`);
+    suite("Full work flow tests", function () {
+        let folderContext: FolderContext;
+        let workspaceContext: WorkspaceContext;
+        const windowMock = mockGlobalObject(vscode, "window");
+
+        suiteSetup(async function () {
+            workspaceContext = await globalWorkspaceContextPromise;
+            await waitForNoRunningTasks();
+            folderContext = await folderContextPromise("dependencies");
+            await workspaceContext.focusFolder(folderContext);
+        });
+
+        test("Use local dependency", async () => {
+            // Expect to fail without setting up local version
+            const tasks = (await getBuildAllTask(folderContext)) as SwiftTask;
+            let { exitCode, output } = await executeTaskAndWaitForResult(tasks);
+            expect(exitCode).to.not.equal(0);
+            expect(output).to.include("PackageLib");
+            expect(output).to.include("required");
+
+            // Contract: spm reset, resolve, update should work
+            const executeTaskSpy = sinon.spy(utilities, "executeTaskWithUI");
+            await resolveDependencies(workspaceContext);
+            await expect(executeTaskSpy.returnValues[0]).to.eventually.be.true;
+
+            await updateDependencies(workspaceContext);
+            await expect(executeTaskSpy.returnValues[1]).to.eventually.be.true;
+
+            await resetPackage(workspaceContext);
+            await expect(executeTaskSpy.returnValues[2]).to.eventually.be.true;
+            await expect(executeTaskSpy.returnValues[3]).to.eventually.be.true;
+
+            // Contract: spm edit with user supplied local version of dependency
+            windowMock.showOpenDialog.resolves([testAssetUri("Swift-Markdown")]);
+            const id = "swift-markdown";
+            await useLocalDependency(id, workspaceContext);
+            await expect(executeTaskSpy.returnValues[4]).to.eventually.be.true;
+
+            // This will now pass as we have the required library
+            ({ exitCode, output } = await executeTaskAndWaitForResult(tasks));
+            expect(exitCode).to.equal(0);
+            expect(output).to.include("defaultpackage");
+            expect(output).to.include("not used by any target");
+
+            // Contract: spm unedit
+            const updateWorkspaceSpy = sinon.spy(vscode.workspace, "updateWorkspaceFolders");
+            // We would love to call uneditDependency for coverage but there's no clean way to get
+            // a synchronize point for deterministic task completion so just call this function direct
+            await uneditFolderDependency(workspaceContext.currentFolder!, id, workspaceContext);
```

Comment:
> "full workflow", i.e. multiple paths and actions, is more smoke level. Find to group a feature in a suite, but > let's make sure each test case is one path.

This is sort of what I meant. I'm not quite sure what we're testing here. We're checking exit codes above, further actions and assertions in subsequent steps. Ideally 1 action, 1 result, 1 set of assertion(s) on that result for each test case

Reply:
Done.